superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterBarCrossFilters/CrossFilter.tsx

```diff
@@ -0,0 +1,229 @@
+          `
+          : `
+            margin-left: ${theme.gridUnit * 2}px;
+          `}
+      `}
+      closable
+      onClose={() => removeCrossFilter(filter.emitterId)}
+    >
+      <Tooltip title={columnIsTruncated ? filter.column : null}>
+        <StyledCrossFilterColumn ref={columnRef}>
+          {filter.column}
+        </StyledCrossFilterColumn>
+      </Tooltip>
+      <Tooltip title={valueIsTruncated ? filter.value : null}>
+        <StyledCrossFilterValue ref={valueRef}>
+          {filter.value}
+        </StyledCrossFilterValue>
+      </Tooltip>
+    </Tag>
+  );
+};
+
+const CrossFilterChartTitle = (props: {
+  title: string;
+  orientation: FilterBarOrientation;
+}) => {
+  const { title, orientation } = props;
+  const [titleRef, titleIsTruncated] = useCSSTextTruncation<HTMLSpanElement>();
+  const theme = useTheme();
+  return (
+    <Tooltip title={titleIsTruncated ? title : null}>
+      <span
+        css={css`
+          max-width: ${orientation === FilterBarOrientation.VERTICAL
+            ? `${theme.gridUnit * 60}px`
+            : `${theme.gridUnit * 15}px`};
+          line-height: 20px;
+          ${ellipsisCss}
+        `}
+        ref={titleRef}
+      >
+        {title}
+      </span>
+    </Tooltip>
+  );
+};
+
+const CrossFilter = (props: {
+  filter: CrossFilterIndicator;
+  orientation: FilterBarOrientation;
+  last?: boolean;
+}) => {
+  const { filter, orientation, last } = props;
+  const theme = useTheme();
+  const dispatch = useDispatch();
+
+  const onHighlightFilterSource = useCallback(
+    (path?: string[]) => {
+      if (path) {
+        dispatch(setFocusedNativeFilter(path[0]));
+      }
+    },
+    [dispatch],
+  );
+
+  const handleRemoveCrossFilter = (chartId: number) => {
+    dispatch(
+      updateDataMask(chartId, {
+        extraFormData: {
+          filters: [],
+        },
+        filterState: {
+          value: null,
+          selectedValues: null,
+        },
+      }),
+    );
+  };
+
+  return (
+    <div
+      key={`${filter.name}${filter.emitterId}`}
+      css={css`
+        ${orientation === FilterBarOrientation.VERTICAL
+          ? `
+            display: block;
+            margin-top: ${theme.gridUnit * 4}px;
+          `
+          : `
+            display: flex;
+          `}
+      `}
+    >
+      <StyledCrossFilterTitle
+        onClick={() => onHighlightFilterSource(filter.path)}
+        role="button"
+        tabIndex={0}
+      >
+        <CrossFilterChartTitle
+          title={filter.name}
+          orientation={orientation || FilterBarOrientation.HORIZONTAL}
+        />
+        <Tooltip title={t('Locate the chart')}>
+          <StyledIconSearch iconSize="s" />
+        </Tooltip>
+      </StyledCrossFilterTitle>
+      {(filter.column || filter.value) && (
+        <CrossFilterTag
+          filter={filter}
+          orientation={orientation}
+          removeCrossFilter={handleRemoveCrossFilter}
+        />
+      )}
+      {last && (
```

Comment:
I'm wondering if that separator line should be a part of `CrossFilter` component 🤔 Maybe we could extract it from here, render all cross filters in a container and then draw that line after the container? That would also let us get rid of that `last` logic

Reply:
The reason behind this choice is that when the cross filters are displayed in horizontal mode, we need to account for both displaying them in the filter bar as well as in "more filters" dropdown. Both use the component directly to display each filter similarly to what we do with native filters. Creating a container around this logic would make the rendering function more complex and introduce a different pattern in the filter controls.